src/tui.rs

```diff
@@ -1734,13 +1730,21 @@ impl<'a> Context<'a, '_> {
 
         // Consume the input unconditionally, so that the root (the "main window")
         // doesn't accidentally receive any input via `consume_shortcut()`.
+        let mut exit = false;
         if self.contains_focus() {
-            let exit = !self.input_consumed && self.input_keyboard == Some(vk::ESCAPE);
-            self.set_input_consumed_unchecked();
-            exit
-        } else {
-            false
+            if let Some(input) = self.input_keyboard {
+                if !self.input_consumed && input == vk::ESCAPE {
+                    exit = true;
+                }
```

Comment:
This is the same as the previous code. Do you mind reverting it to this?
```rs
let exit = !self.input_consumed && self.input_keyboard == Some(vk::ESCAPE);
```

You can then also remove the `exit` variable above.